lib/rule-data-snapshot.js

```diff
@@ -3,6 +3,14 @@
 
 var merge = require('lodash.mergewith');
 
+function isEmptyObj (obj) {
+  return obj && typeof obj == 'object' && Object.keys(obj).length === 0 && obj.constructor === Object;
```

Comment:
per [here](http://stackoverflow.com/questions/679915/how-do-i-test-for-an-empty-javascript-object), a `new Date()` would also pass as an empty object without the extra check.